src/hooks/useOfflineDetector.tsx

```diff
@@ -0,0 +1,70 @@
+import { useEffect, useRef, useState } from 'react';
+
+const POLLING_DELAY = 5000;
+
+export type UseOfflineDetectorParams = {
+    setStatus: (online: boolean) => void;
+};
+
+export const useOfflineDetector = ({ setStatus }: UseOfflineDetectorParams) => {
+    /** General status */
+    const [onlineGlobal, setOnlineGlobal] = useState<boolean>(true);
```

Comment:
Type will be provided automatically because you set default value. `<boolean>` is unnecessary here.